### PR TITLE
fix NPE #210 and respond according to spec; propose less intrusive showMessage/Request() ui

### LINKED ISSUE #210: showMessageRequest fail if no actions are set
**Description:**

DefaultLanguageClient.showMessageRequest fails and throws an NPE if no actions are send in the request. Such a request is perfectly valid. In this case, the user should be prompted with an "OK" prompt displaying the message.

See interface for ShowMessageRequestParams here: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#window_showMessageRequest

**Suggested Labels:**

Type/Bug

**Suggested Assignees:**

**Affected Product Version:**
0.94.2

**OS, DB, other environment details and versions:**    
Windows 10, Intellij 2020.1.1

**Steps to reproduce:**
Have the language server send a ShowMessageRequestParams object that does not have any actions. Observe NPE.

**Related Issues:**

### PR DESCRIPTION
**Pull request by swissiety**
## Purpose
1. closes #210 - checks whether actions are received (i.e. are null) and not assuming that returns an empty list
2. responds with null if the dialog/notification is closed without a choosing an action - according to LSP Spec (3.15).
3. Additionally this PR introduces the ability to make showMessage/+Request less intrusive via non-modal Notifications.

### CODE REVIEW

File: src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java [hunk 1]
```diff
@@ -126,60 +120,131 @@ public void publishDiagnostics(PublishDiagnosticsParams publishDiagnosticsParams
...
+                    case Info:
+                    case Log:
+                        Messages.showInfoMessage(message, title);
+                        break;
+                    default:
+                        LOG.warn("No message type for " + message);
+                        break;
+                }
+            });
+
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java:143]: Redundant blank line?

File: src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java [hunk 2]
```diff
@@ -126,60 +120,131 @@ public void publishDiagnostics(PublishDiagnosticsParams publishDiagnosticsParams
...
+        if (isModal) {
+            ApplicationUtils.invokeLater(() -> {
+                MessageType msgType = messageParams.getType();
+                switch (msgType) {
+                    case Warning:
+                        Messages.showWarningDialog(message, title);
+                        break;
+                    case Error:
+                        Messages.showErrorDialog(message, title);
+                        break;
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java:133]: Shall we consider swapping `Warning` and `Error` cases, just be aligned with the java log level standards?

File: src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java [hunk 3]
```diff
@@ -126,60 +120,131 @@ public void publishDiagnostics(PublishDiagnosticsParams publishDiagnosticsParams
...
-        return CompletableFuture.completedFuture(new MessageActionItem(actions.get(exitCode).getTitle()));
+    @NotNull
+    protected NotificationType getNotificationType(@NotNull MessageType messageType) {
+        NotificationType type;
+        switch (messageType) {
+            case Warning:
+                type = NotificationType.WARNING;
+                break;
+            case Error:
+                type = NotificationType.ERROR;
```
Comment (NipunaRanasinghe) [inline src/main/java/org/wso2/lsp4intellij/client/DefaultLanguageClient.java:236]: Shall we consider swapping `Warning` and `Error` cases, just be aligned with the java log level standards?

**Review by NipunaRanasinghe — COMMENTED** [review]
@swissiety Really appreciate your time on this.. LGTM except few minor suggestions that I've added.

**Comment by NipunaRanasinghe** [comment]
@swissiety I've also noticed that this PR seems to be failing due to spotbugs issues. Shall we get them fixed too?

**Comment by swissiety** [comment]
> @swissiety I've also noticed that this PR seems to be failing due to spotbugs issues. Shall we get them fixed too?

how can i access the spotbugs report?

**Comment by NipunaRanasinghe** [comment]
> > @swissiety I've also noticed that this PR seems to be failing due to spotbugs issues. Shall we get them fixed too?
> 
> how can i access the spotbugs report?

@swissiety Try running `gradle build` command locally. Then it should fail with spotbug failures. You can access the report in there(You'll get an url for the spotbug report).

**Review by NipunaRanasinghe — APPROVED** [review]
Looks good!

**Pull request merged**